src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,107 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+const isValid = (value: string): boolean => {
+  return value.trim().length > 0;
+};
+
+enum Fields {
+  title = 'Title',
+  description = 'Description',
+  imgUrl = 'Image URL',
+  imdbUrl = 'Imdb URL',
+  imdbId = 'Imdb ID',
+}
+
+export const NewMovie: React.FC<Props> = props => {
+  const { onAdd } = props;
```

Comment:
```suggestion
export const NewMovie: React.FC<Props> = { onAdd } => {
```

can be a bit simplify